bio2zarr/vcf2zarr/vcz.py

```diff
@@ -463,6 +514,29 @@ def fromdict(d):
         return ret
 
 
+def compute_laa_field(genotypes) -> np.ndarray:
+    """
+    Computes the value of the LAA field for each sample given the genotypes
+    for a variant.
+
+    The LAA field is a list of one-based indices into the ALT alleles
+    that indicates which alternate alleles are observed in the sample.
+    """
+    v = 2**31 - 1
+    if np.any(genotypes >= v):
+        raise ValueError("Extreme allele value not supported")
+    G = genotypes.astype(np.int32)
+    # Anything <=0 gets mapped to -2 (pad) in the output, which comes last.
+    # So, to get this sorting correctly, we remap to the largest value for
+    # sorting, then map back. We promote the genotypes up to 32 bit for convenience
+    # here, assuming that we'll never have a allele of 2**31 - 1.
+    assert np.all(G != v)
+    G[G <= 0] = v
+    G.sort(axis=1)
+    G[G == v] = -2
```

Comment:
Note: we've missed a case here where the two values are equal and we should have the pad element also.